From d777342610280819c7c59816b258cebc9f7ca2f1 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 16 Nov 2014 10:44:59 +0000
Subject: [PATCH] MythUIHelper: use the theme search paths to find the widget
 background image

This allows a theme to fallback to using a 'BackgroundPixmap' or
'TiledBackgroundPixmap' image from a base theme.

We now also warn if one wasn't found.
---
 mythtv/libs/libmythui/mythuihelper.cpp |   13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/mythtv/libs/libmythui/mythuihelper.cpp b/mythtv/libs/libmythui/mythuihelper.cpp
index b9c791a..83a2a27 100644
--- a/mythtv/libs/libmythui/mythuihelper.cpp
+++ b/mythtv/libs/libmythui/mythuihelper.cpp
@@ -1291,8 +1291,10 @@ void MythUIHelper::ThemeWidget(QWidget *widget)
 
     if (!d->m_qtThemeSettings->GetSetting("BackgroundPixmap").isEmpty())
     {
-        QString pmapname = d->m_themepathname +
-                           d->m_qtThemeSettings->GetSetting("BackgroundPixmap");
+        QString pmapname = d->m_qtThemeSettings->GetSetting("BackgroundPixmap");
+        if (!FindThemeFile(pmapname))
+            LOG(VB_GENERAL, LOG_ERR, QString(LOC + "Failed to find '%1' in the theme search path")
+                .arg(d->m_qtThemeSettings->GetSetting("BackgroundPixmap")));
 
         bgpixmap = LoadScalePixmap(pmapname);
 
@@ -1305,8 +1307,11 @@ void MythUIHelper::ThemeWidget(QWidget *widget)
     else if (!d->m_qtThemeSettings
              ->GetSetting("TiledBackgroundPixmap").isEmpty())
     {
-        QString pmapname = d->m_themepathname +
-                           d->m_qtThemeSettings->GetSetting("TiledBackgroundPixmap");
+        QString pmapname = d->m_qtThemeSettings->GetSetting("TiledBackgroundPixmap");
+
+        if (!FindThemeFile(pmapname))
+            LOG(VB_GENERAL, LOG_ERR, QString(LOC + "Failed to find '%1' in the theme search path")
+                .arg(d->m_qtThemeSettings->GetSetting("TiledBackgroundPixmap")));
 
         int width, height;
         float wmult, hmult;
-- 
1.7.10.2

